qiskit/primitives/base/base_estimator.py

```diff
@@ -254,3 +346,44 @@ def parameters(self) -> tuple[ParameterView, ...]:
             Parameters, where ``parameters[i][j]`` is the j-th parameter of the i-th circuit.
         """
         return tuple(self._parameters)
+
+
+BaseEstimator = BaseEstimatorV1
+
+
+class BaseEstimatorV2(BasePrimitiveV2, Generic[T]):
+    """Estimator base class version 2.
+
+    Estimator estimates expectation values of quantum circuits and observables.
+    """
+
+    def __init__(self, options: Optional[BasePrimitiveOptionsLike]):
+        super().__init__(options=options)
+
+    def run(self, tasks: EstimatorTaskLike | Iterable[EstimatorTaskLike]) -> T:
+        """Run the tasks of the estimation of expectation value(s).
+
+        Args:
+            tasks: a tasklike object. Typically, list of tuple
+                ``(QuantumCircuit, observables, parameter_values)``
+
+        Returns:
+            The job object of Estimator's Result.
+        """
+        if isinstance(tasks, EstimatorTask):
```

Comment:
Do we really want all this task coercion and validation in the base class or just leave it as an abstract interface? There is no reason the body of this method couldn't just be a utility function someone could call in their primitive implementation since it has no dependence on `self`. I would remove it and just have 
```python
@abstractmethod
def run(self, tasks: EstimatorTaskLike | Iterable[EstimatorTaskLike]) -> T:
```

And then an Estimator implementation could implement the body as

```
tasks = coerce_estimator_tasks(tasks)
# run logic...
```